code/modules/admin/callproc/callproc.dm

```diff
@@ -139,8 +139,9 @@
 				if(isnull(current)) return CANCEL
 
 			if("type")
-				current = input("Select type for [arguments.len+1]\th argument") as null|anything in typesof(/obj, /mob, /area, /turf)
+				current = input("Enter type path for [arguments.len+1]\th argument") as null|text
 				if(isnull(current)) return CANCEL
+				current = text2path(current)
```

Comment:
You need to ispath check this (and ispath(current, /atom) possibly, as you could inject other paths), as text2path has weird fail behavior. On older BYOND versions you'd have to do this in a try-catch block, but I believe that's no longer the case.

Reply:
The big reason I added that was to be able to pass datums, sooo. Also that's a debugging tool, which expects whoever uses it to know what they're doing with it?

Reply:
Passing datums is fine, just do `ispath(foo)` before proceeding in case they didn't enter a path at all. I agree it doesn't need to be super safe, but at the same time no reason to cause unnecessary runtimes when the user misspells things. Can even to_chat them that the path wasn't found or whatever.

Reply:
Alright then. I'll do that!